### PR TITLE
doc: add Nix documentation

### PR DESCRIPTION
**Pull request by imincik**
Add instructions how to install Nix, use it to create a development environment and install GRASS from git.  

New file is created because this information doesn't fit well enywhere else (I spent some time with @veroandreo trying to find a suitable space).

/cc @veroandreo

### CODE REVIEW

**Comment by echoix** [comment]
I'm ok with this! I'll just let @veroandreo take a look for the contents and wording, but you have my approval when it will be needed and ready

**Review by echoix — COMMENTED** [review]
As I think people that would read docs are less likely to know a lot about it and maybe less technical, I suggest having the most useful instructions for a user near the top (in a logical order still), and having the developer docs more at the end, where it is more advanced info

**Comment by imincik** [comment]
> As I think people that would read docs are less likely to know a lot about it and maybe less technical, I suggest having the most useful instructions for a user near the top (in a logical order still), and having the developer docs more at the end, where it is more advanced info

Thank you for review. I think, this file contains only the most important information and the order looks good to me as well. Do you have something specific  in mind ?

**Comment by echoix** [comment]
Is it possible to just run   ```bash
  nix run github:OSGeo/grass#grass
  ```
Without having the direnv set?
What's the recommended way to install nix for you?

File: NIX.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# How to use Nix
+
+## Nix installation
```
Comment (neteler) [inline NIX.md:3]:
```suggestion
## What is Nix

[Nix](https://nixos.org/) is a powerful package manager and system configuration tool
that aims to make software deployment fully reproducible.

## Nix installation
```

One intro sentence will not harm.

File: NIX.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+- Install Nix
+  [(learn more about this installer)](https://zero-to-nix.com/start/install)
+
+  ```bash
+  curl --proto '=https' --tlsv1.2 -sSf \
+     -L https://install.determinate.systems/nix \
+     | sh -s -- install
+  ```
+
+## Create GRASS development environment
```
Comment (neteler) [inline NIX.md:14]:
```suggestion
## Create GRASS GIS development environment
```

File: NIX.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+- Optionally, use [direnv](https://direnv.net) to activate environment
+  automatically when entering the source code directory
+
+  ```bash
+  echo "use flake" > .envrc
+  direnv allow
+  ```
+
+## Launch GRASS directly from the source code
```
Comment (neteler) [inline NIX.md:32]:
```suggestion
## Launch GRASS GIS directly from the source code
```

File: NIX.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  nix run github:OSGeo/grass#grass
+  ```
+
+- Launch GRASS from specific Git revision, branch or tag
+
+  ```bash
+  nix run github:OSGeo/grass/<REVISION|BRANCH|TAG>#grass
+  ```
+
+## Install GRASS directly from the source code
```
Comment (neteler) [inline NIX.md:54]:
```suggestion
## Install GRASS GIS directly from the source code
```

File: NIX.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  nix profile install github:OSGeo/grass#grass
+  ```
+
+- Install GRASS from specific Git revision, branch or tag
+
+  ```bash
+  nix profile install github:OSGeo/grass/<REVISION|BRANCH|TAG>#grass
+  ```
+
+## Uninstall GRASS
```
Comment (neteler) [inline NIX.md:75]:
```suggestion
## Uninstall GRASS GIS
```

**Comment by neteler** [comment]
Shouldn't this rather go into the `doc/` subdir?

**Comment by echoix** [comment]
> Thanks for writing this up, @imincik!
> 
> 
> 
> Question: shouldn't this rather go into the `doc/` subdir?

I think so too, unless there is a way stronger convention in the nix ecosystem, but a doc subfolder is an universal convention too.

File: NIX.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+  ```bash
+  nix run github:OSGeo/grass#grass
+  ```
+
+- Launch GRASS from specific Git revision, branch or tag
+
+  ```bash
+  nix run github:OSGeo/grass/<REVISION|BRANCH|TAG>#grass
+  ```
```
Comment (marisn) [inline NIX.md:52]: Could you, please, also add an example how to run a GRASS PR?

**Comment by imincik** [comment]
Thanks for feedback @neteler and @echoix . File is now moved to `doc` directory.

**Comment by imincik** [comment]
> Is it possible to just run
> 
> ```shell
>   nix run github:OSGeo/grass#grass
> ```

> Without having the direnv set?

Yes, actually direnv is not useful for `nix run`. It is mostly used to activate development environment instead of manually running `nix develop`.


> What's the recommended way to install nix for you?

As documented in `NIX.md` - use Determinate Systems installer (this is the company where the founder of Nix works). I think, this installer is safer and enables some features which we use for GRASS.

File: NIX.md (hunk 6 above)
Comment (imincik) [reply NIX.md]: There is no more magic, just run `nix run github:<YOUR-GITHUB-NAME>/<GRASS>/<PR-BRANCH>#grass`. Maybe it makes a sense to document it explicitly.

Comment (imincik) [reply NIX.md]: Done.

**Review by echoix — APPROVED** [review]

**Pull request merged**